rnd/autogpt_libs/autogpt_libs/supabase_integration_credentials_store/types.py

```diff
@@ -0,0 +1,51 @@
+from uuid import uuid4
+from typing import Annotated, Any, Literal, Optional, TypedDict
+
+from pydantic import BaseModel, Field, SecretStr, field_serializer
+
+
+class _BaseCredentials(BaseModel):
+    id: str = Field(default_factory=lambda: str(uuid4()))
+    provider: str
+    title: str
+
+    @field_serializer("*")
+    def dump_secret_strings(value: Any, _info):
+        if isinstance(value, SecretStr):
+            return value.get_secret_value()
+        return value
+
+
+class OAuth2Credentials(_BaseCredentials):
+    type: Literal["oauth2"] = "oauth2"
+    access_token: SecretStr
+    access_token_expires_at: int
+    refresh_token: SecretStr
+    refresh_token_expires_at: Optional[int]
+    scopes: list[str]
+
+
+class APIKeyCredentials(_BaseCredentials):
+    type: Literal["api_key"] = "api_key"
+    api_key: SecretStr
+    expires_at: Optional[int]
+
+
+class PasswordCredentials(_BaseCredentials):
```

Comment:
I think the name should be BasicAuthCredentials - wdyt?

Reply:
Basic auth would hint at the protocol that will be used for the actual authentication step. Is Basic the only protocol for username+password auth though? I guess not but I don't know.

Reply:
It's not you're right but I can't think of a single integration that wouldn't use basic auth for username password 

(naming is a nit btw, i just can't approve this PR since it's mine)

Reply:
Well there's stuff like:
* LDAP
* SMTP (already have a block for that)
* (S)FTP

But those also require a URL and maybe some other stuff to complete the set of login info. I'll just remove these extra credential types and we can add them back as needed.